code/datums/components/doorhonk.dm

```diff
@@ -0,0 +1,27 @@
+/datum/component/doorhonk
+
+	var/honkSound = 'sound/items/bikehorn.ogg'
+	var/volume = 30
+	var/honkLeft
+	var/infinite = FALSE
+
+/datum/component/doorhonk/Initialize(volume_override = 30, infinite_honking = FALSE)
+	if(!istype(parent, /obj/machinery/door/airlock)) //This component only works for doors.
+		return COMPONENT_INCOMPATIBLE
+
+	honkLeft = (rand(15,20)) //This component works for an amount of honk between 15 and 20
```

Comment:
Include this in the parameters. Something like this:
`/datum/component/doorhonk/Initialize(volume_override = 30, honk_amount = rand(15, 20))`
Then you can add something like `if(honk_amount <= 0) infinite = TRUE`

Reply:
Doesn't work, can't call rand in parameters.